dp/generate_prsn.py

```diff
@@ -222,6 +229,11 @@ def process_to_prsn(variables, output_dataset, max_chunk_len):
             }
         means = np.mean([chunk_data['tasmin'], chunk_data['tasmax']], axis=0)
         prsn_data = np.where(means < freezing, chunk_data['pr'], 0)
+
+        prsn_units_from = ureg.parse_units('kg/m^2/s')
+        prsn_units_to = ureg.parse_units('g/cm^2/s')
+        prsn_data = prsn_data * Q_(1.0, prsn_units_from).to(prsn_units_to).magnitude	
+
         output_dataset.variables['prsn'][chunk] = prsn_data
```

Comment:
Have you considered skipping the extra variable assignment and assigning directly?

```python3
output_dataset.variables['prsn'][chunk] = prsn_data * Q_(1.0, prsn_units_from).to(prsn_units_to).magnitude
```